### PR TITLE
fix: ms sql default port updated to 1433

### LINKED ISSUE #16808: [Bug]-[3240]:5432 is considered the default port if port is not provided in MSSQL
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The default port for `MSSQL` database is `1433` but in the source code I’ve found that it’s `5432` which is the default for `Postgres`.

### Steps To Reproduce

1. Go to Datasources -> Create New
2. Choose Microsoft SQL Server as the datasource
3. Provide all the information in the datasource-create page
4. Leave the port blank

### Public Sample App

_No response_

### Version

Cloud / Self Hosted

### PR DESCRIPTION
**Pull request by vaibh1297**
## Description

- Ms sql default port updated to 1433

Fixes #16808

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Unit Testing

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/mssqlPlugin/src/main/java/com/external/plugins/MssqlPlugin.java [hunk 1]
```diff
@@ -526,6 +526,15 @@ public Object substituteValueInInput(int index,
         }
     }
 
+    public static long getPort(Endpoint endpoint) {
+
+        if (endpoint.getPort() == null) {
+            return 1433L;
```
Comment (sumitsum) [inline app/server/appsmith-plugins/mssqlPlugin/src/main/java/com/external/plugins/MssqlPlugin.java:532]: Can you please replace the constant by a macro e.g. `MSSQL_DEFAULT_PORT` ?

**Comment by sumitsum** [comment]
Also, another point for future PRs : please name the PR branches as `fix/my_fix_desc` or `feature/my_feature_desc` as described here: https://github.com/appsmithorg/appsmith/blob/release/contributions/CodeContributionsGuidelines.md

**Comment by btsgh** [comment]
Trying to test this by connecting through a docker container of ms sql, but am stuck at the point where the image is not starting up as image platform (linux/amd64) does not match my host system platform (linux/arm64/v8)

**Comment by btsgh** [comment]
Still stuck on issues with setting up docker container image for ms sql database. In conversation with Devteam with regards the same. (There are no local instances of MSSQL available to connect to and test)

**Comment by btsgh** [comment]
Tested the issue on localhost with an MSSQL docker container image - 
1. Test connection to the container image without the port (field blank) specified passes.
2. Test connection to the container image with the correct port specified passes.
3. Test connection to the container image with the incorrect port specified does not go through. [No error message either]
4. Test connection to the container image with the incorrect db name/username/password gives error message - as expected. 
5. Saving the datasource with the port field left blank passes, and also running a sample query fetches records from the db. [SELECT
  * FROM SYSOBJECTS WHERE xtype = 'U';]

**Pull request merged**